opal/util/output.h

```diff
@@ -552,6 +554,30 @@ OPAL_DECLSPEC void opal_output_set_output_file_info(const char *dir, const char
  */
 OPAL_DECLSPEC OBJ_CLASS_DECLARATION(opal_output_stream_t);
 
+/**
+ * Best effort write for blocking file descriptors
+ *
+ * A wrapper around write() that will spin trying to write the entire
+ * buffer until either an error occurs or a write is not able to write
+ * any data.  Useful for situations where opal_output cannot be used
+ * for signal reasons.  Do not use with non-blocking file descriptors.
+ */
+static inline int opal_best_effort_write(int fd, const void *buf, size_t count)
+{
+    size_t written = 0;
+
+    while (written != count) {
+        ssize_t tmp = write(fd, (char *)buf + written, count - written);
+        /* we explicitly do not recover from errors in this wrapper */
+        if (tmp <= 0) {
```

Comment:
Could use `if (tmp < 0)` here as 0 is a "successful write of 0 bytes".  Errors will return `-1` and set errno.

Could also try again error in the case of `EAGAIN`.

Reply:
It's unspecified what to do with 0, given that frequently fd is not a regular file, which is why I treated it as an error for this case.  It's not clear it would be safe to retry, with any guarantee of progress.

EAGAIN can't happen on a write to a blocking fd, and I was pretty clear in the documentation that this was for blocking fds.

Reply:
Good point, EINTR can happen with a blocking socket.  The chance of it happening without writing a single byte is pretty small, but non-zero.  So updated the patch to handle that case.